src/components/table/TableCell.js

```diff
@@ -218,4 +220,6 @@ class TableCell extends PureComponent {
   }
 }
 
-export default TableCell;
+export default connect(state => ({
+  modalVisible: state.windowHandler.modal.visible,
```

Comment:
Can't we grab this in one of the parent containers that is already connected to the store ? We should limit the amount of components subscribing to the store and create containers that handle data fetching instead.

Reply:
Good catch, that piece of code is definitively worth talking about.

Since it's an interesting argument, I hope you don't mind if I elaborate a bit. Generally I like how `connect` makes a component self-contained. It works really well, especially if the usage of the component is always the same.

Consider the following example:

```js
// TableCell expects a "modalVisible" prop. It doesn't really tell a good story of why it needs it

ComponentA(props) {
  return <TableCell modalVisible={props.modalVisible} />;
}

ComponentB(props) {
  return <TableCell modalVisible={props.modalVisible} />;
}

ComponentC(props) {
  return <TableCell modalVisible={props.modalVisible} />;
}
```

vs

```js
// TableCell is connected to "modalVisible" and does not concern its parent components about it

ComponentA() {
  return <TableCell />;
}

ComponentB() {
  return <TableCell />;
}

ComponentC() {
  return <TableCell />;
}
```

Also performance-wise I'd argue that connecting `TableCell` is better. If `modalVisible` changes, only the cells will re-render, since ultimatively their render output is dependent on that value. If you connect parent components, they will re-render without having changed anything, only for the purpose to provide `TableCell` with a new value.

> https://github.com/reactjs/redux/blob/master/docs/faq/Performance.md
> [...] many individual components should be connected to the store instead of just a few, and connected list components should pass item IDs to their connected child list items (allowing the list items to look up their own data by ID). This minimizes the overall amount of rendering to be done.

---

On a side note: Regarding the container/component debate, you might be interested in reading this:

https://twitter.com/dan_abramov/status/802569801906475008

Reply:
Well yeah, normally I would agree. In this case though : 
- it's just a single prop, so doesn't change much. Unless we'll define TableCell props as {...props} instead of defining them separately, it won't really matter.
- I agree that there's no reason to be anal about component/container definition and separation. The thing is though our components are really complicated and often 1000 lines long. So in this case, rewriting them into two files would help with maintainability and readability.

Last but not least, quoting Dan:
```
Components often flipped back and forth as we worked on them.
```

Reply:
I'm all about changing things as we go.

So if I understand correctly -  you want to wrap a component in a container that's doing the data fetching and then reusing that container instead of the component directly?

Yes, that would be perfectly valid to seperate the requirements of our larger components. We just need to care that not too much logic creeps into those.

For this component: I suppose you agree it doesn't make sense to introduce a container that fetches one prop - fine to keep it like that now?

Reply:
Yup, for this component it makes no sense.